crates/kernel/src/tasks/scheduler.rs

```diff
@@ -1327,98 +1290,32 @@ impl Scheduler {
 
         // Then abort all of their still-living forked tasks (that weren't the disconnect
         // task, we need to let that run to completion for sanity's sake.)
-        for (task_id, task) in inner.tasks.iter() {
+        for (task_id, tc) in self.tasks.iter() {
             if *task_id == disconnect_task_id {
                 continue;
             }
-            if task.player != player {
+            if tc.player != player {
                 continue;
             }
             warn!(
                 ?player,
                 task_id, "Aborting task from disconnected player..."
             );
-            // This is fire and forget, we cannot assume that the task is still alive.
-            let tcs = task.task_control_sender.clone();
-            let Ok(_) = tcs.send(TaskControlMsg::Abort) else {
-                trace!(?player, task_id, "Task already dead");
-                continue;
-            };
+            tc.kill_switch.store(true, Ordering::SeqCst);
         }
-    }
-
-    fn process_task_removals(&self, to_remove: &[TaskId]) {
-        let mut inner = self.inner.write().unwrap();
+        // Likewise, suspended tasks.
+        let to_remove = self
+            .suspended
+            .iter()
+            .filter_map(|(task_id, sr)| {
```

Comment:
Is it possible for the disconnect thread to be suspended?

Reply:
Yes, there's a whole bunch of disconnect cases we're going to have to potentially deal with. 

How LambdaMOO deals with them, is a mystery to me right now.